tasks/deploy.py

```diff
@@ -0,0 +1,445 @@
+
+    return slurm_out
+
+
+def determine_eessi_tarballs(job_dir):
+    """Determine EESSI tarballs.
+
+    Args:
+        job_dir (string): directory of a job
+
+    Returns:
+        eessi_tarballs (list): list of paths to all tarballs in job_dir
+    """
+    # determine all tarballs that are stored in the directory job_dir
+    #   and whose name matches a certain pattern
+    tarball_pattern = "eessi-*software-*.tar.gz"
+    glob_str = os.path.join(job_dir, tarball_pattern)
+    eessi_tarballs = glob.glob(glob_str)
+
+    return eessi_tarballs
+
+
+def check_build_status(slurm_out, eessi_tarballs):
+    """Check status of the job in a given directory.
+
+    Args:
+        slurm_out (string): path to slurm*out file
+        eessi_tarballs (list): list of eessi tarballs found for job
+
+    Returns:
+        (bool): True -> job succeeded, False -> job failed
+    """
+    # Function checks if all modules have been built and if a tarball has
+    # been created.
+
+    # set some initial values
+    no_missing_modules = False
+    targz_created = False
+
+    # check slurm out for the below strings
+    #   ^No missing modules!$ --> all software successfully installed
+    #   ^/eessi_bot_job/eessi-.*-software-.*.tar.gz created!$ -->
+    #     tarball successfully created
+    if os.path.exists(slurm_out):
+        re_missing_modules = re.compile("^No missing modules!$")
+        re_targz_created = re.compile("^/eessi_bot_job/eessi-.*-software-.*.tar.gz created!$")
+        outfile = open(slurm_out, "r")
+        for line in outfile:
+            if re_missing_modules.match(line):
+                # no missing modules
+                no_missing_modules = True
+            if re_targz_created.match(line):
+                # tarball created
+                targz_created = True
+
+    # we test results from the above check and if there is one tarball only
+    if no_missing_modules and targz_created and len(eessi_tarballs) == 1:
+        return True
+
+    return False
+
+
+def update_pr_comment(tarball, repo_name, pr_number, state, msg):
+    """Update PR comment which contains specific tarball name.
+
+    Args:
+        tarball (string): name of tarball that is looked for in a PR comment
+        repo_name (string): name of the repository (USER_ORG/REPOSITORY)
+        pr_number (string): pull request number
+        state (string): state (upload) to be used in update
+        msg (string): msg (succeeded or failed) describing upload result
+    """
+    funcname = sys._getframe().f_code.co_name
+
+    gh = github.get_instance()
+    repo = gh.get_repo(repo_name)
+    pull_request = repo.get_pull(pr_number)
+
+    # TODO does this always return all comments?
+    comments = pull_request.get_issue_comments()
+    for comment in comments:
+        # NOTE
+        # adjust search string if format changed by event handler
+        # (separate process running eessi_bot_event_handler.py)
+        re_tarball = f".*{tarball}.*"
```

Comment:
This pattern should be something we import from the event handler, perhaps?

Or the job manager and this task should use the same function to update the status in the appropriate PR comment

Reply:
See issues #108 and #109